### PR TITLE
AVRO-1743: override writeFixed in BlockingBinaryEncoder

### PR DESCRIPTION
**Pull request by tbobrovytsky**

### CODE REVIEW

File: lang/java/avro/src/main/java/org/apache/avro/io/BlockingBinaryEncoder.java [hunk 1]
```diff
@@ -287,7 +288,20 @@ public void writeDouble(double d) throws IOException {
   public void writeFixed(byte[] bytes, int start, int len) throws IOException {
     doWriteBytes(bytes, start, len);
   }
-  
+
+  @Override
+  public void writeFixed(ByteBuffer bytes) throws IOException {
+    int pos = bytes.position();
+    int len = bytes.limit() - pos;
```
Comment (rdblue) [inline lang/java/avro/src/main/java/org/apache/avro/io/BlockingBinaryEncoder.java]: Shouldn't this be `bytes.remaining()`?

File: lang/java/avro/src/test/java/org/apache/avro/TestDataFileReflect.java [hunk 2]
```diff
@@ -17,16 +17,19 @@
  */
 package org.apache.avro;
 
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
+import java.io.*;
```
Comment (rdblue) [inline lang/java/avro/src/test/java/org/apache/avro/TestDataFileReflect.java]: Nit: Please don't use wildcard imports.

**Comment by rdblue** [comment]
@tbobrovytsky, can you please write up the problem? It's not very clear what problem this is fixing.

Thanks for working on this!

**Comment by busbey** [comment]
@tbobrovytsky are you still working on this?

**Pull request closed without merging**